src/js/base/js-numbers.js

```diff
@@ -1201,16 +1201,44 @@ define("pyret-base/js/js-numbers", function() {
       return bnRemainder.call(m, n);
     });
 
+  var splitIntintoMantissaExpt = function(s) {
+    var str = s.toString();
+    var c0 = str[0];
+    var sign = '';
+
+
+    if (c0 === '-' || c0 === '+') {
+      str = str.substring(1);
+      if (c0 === '-') {
+        sign = '-';
+      }
+    }
+
+    var d0 = str[0];
+    str = str.substring(1);
+
+    var mantissa = Number(sign + d0 + '.' + str);
+    var expt = str.substring(1).length;
+
+    return [mantissa, expt];
+  };
+
   // _integerDivideToFixnum: integer-pyretnum integer-pyretnum -> fixnum
   var _integerDivideToFixnum = makeIntegerBinop(
     function(m, n) {
       return m / n;
     },
     function(m, n) {
-      return toFixnum(m) / toFixnum(n);
+      var xm = splitIntintoMantissaExpt(m);
+      var xn = splitIntintoMantissaExpt(n);
+      var r = Number(String(xm[0] / xn[0]) + 'e' + 
```

Comment:
Yes, `splitIntIntoMantissaExpt` discards precision (`_integerDivideToFixnum` is all about losing precision in the least lossy way), but so too does `toFixnum`. Using two calls to `splitIntIntoMantissaExpt` instead of two calls to `toFixnum` at least gives a chance for the exponents to cancel out in cases where otherwise overflow would result.